qgis_hub_plugin/gui/resource_browser.py

```diff
@@ -87,6 +94,13 @@ def update_resource_filter(self):
         geopackage_checked = self.checkBoxGeopackage.isChecked()
         style_checked = self.checkBoxStyle.isChecked()
         model_checked = self.checkBoxModel.isChecked()
+        current_text = self.lineEditSearch.text()
+
+        self.checkbox_states = {
+            "Geopackage": geopackage_checked,
+            "Style": style_checked,
+            "Model": model_checked,
+        }
```

Comment:
It's also good to extract this part of the code to a function, something like `update_checkbox_state`. Then you can call it also when calling `__init__`